### PR TITLE
Remove abstract decorator from `_export`

### PR DESCRIPTION
**Pull request by JingyaHuang**
It breaks subclasses which inherit the class without defining `_export`, I would suggest we remove it at least for now.

### CODE REVIEW

**Review by mht-sharma — APPROVED** [review]
Thanks @JingyaHuang, LGTM

**Comment by JingyaHuang** [comment]
I did another commit that I could probably revert, shall we migrate all existing `_from_transformers` to `_export`? @fxmarty @mht-sharma @echarlaix

File: optimum/modeling_base.py [hunk 1]
```diff
@@ -395,7 +394,7 @@ def from_pretrained(
         elif export and trust_remote_code is None:
             trust_remote_code = False
 
-        from_pretrained_method = cls._from_transformers if export else cls._from_pretrained
+        from_pretrained_method = cls._export if export else cls._from_pretrained
```
Comment (fxmarty) [inline optimum/modeling_base.py:397]: This assumes that `_export` has been implemented

Comment (JingyaHuang) [reply optimum/modeling_base.py:397]: yeah, ifwe merge this, we might need to do a sync release for optimum and all sub-packages. Everyone changes.

Comment (fxmarty) [reply optimum/modeling_base.py:397]: okay sounds good. so at this rate we could leave the `@abstractmethod`?

**Comment by JingyaHuang** [comment]
Never mind let's discuss that in another PR.

**Review by echarlaix — APPROVED** [review]

File: optimum/modeling_base.py (hunk 1 above)
Comment (echarlaix) [reply optimum/modeling_base.py]: also the case for optimum-furiosa https://github.com/huggingface/optimum-furiosa/blob/bc77f8408cc5b7ceb66044bd6bc30c38fbf2c568/optimum/furiosa/modeling_base.py#L195 cc @mht-sharma

**Pull request merged**